roles/satellite-clone/tasks/main.yml

```diff
@@ -45,16 +45,33 @@
   fail: msg="Please update the variables in /etc/satellite-clone/satellite-clone-vars.yml"
   when: ((activationkey == "changeme") or (org == "changeme")) and register_to_portal
 
-# Register/subscribe the VM to Red Hat Portal
-- name: register host
+- name: Register/Subscribe the system to Red Hat Portal
   command: "subscription-manager register --force --activationkey='{{ activationkey }}' --org '{{ org }}'"
   when: register_to_portal
-- name: disable all repos
+
+- name: Disable all repos
   command: subscription-manager repos --disable "*"
-  when: register_to_portal
-- name: Enable required repos
+  ignore_errors: True
+  register: disable_repos_result
+  when: enable_repos
+- name: Check that the repos are disabled without errors
+  fail:
+    msg: "Disabling repos failed. Make sure that your system is registered and subscribed to Red Hat. Alternatively,
+         you can skip this task by specifying `enable_repos` to `False`."
+  when: enable_repos and disable_repos_result.rc != 0
```

Comment:
I think you can do `disable_repos_result | success` in ansible now, which is a bit more readable (same for below too)

Reply:
(it may need parenthesis to work with this boolean logic, not sure)